Source/Particles/WarpXParticleContainer.cpp

```diff
@@ -464,9 +474,19 @@ WarpXParticleContainer::DepositCharge (WarpXParIter& pti, RealVector& wp,
     // deposition algorithm are not trivial, this check might be too strict and
     // we might need to relax it, as currently done for the current deposition.
     const int shape_extent = static_cast<int>(WarpX::nox / 2 + 1);
+#ifndef AMREX_USE_GPU
+    // On CPU: Particles deposit on tile arrays,
+    // which have a small number of guard cells
     AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
         amrex::numParticlesOutOfRange(pti, ng_rho - shape_extent) == 0,
-        "Particles shape does not fit within guard cells used for local charge deposition");
+        "Particles shape does not fit within tile used for local charge deposition");
+#else
+    // On GPU: Particles deposit directly on the rho arrays,
+    // which usually have a larger number of guard cells
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
+        amrex::numParticlesOutOfRange(pti, rho->nGrow().min() - shape_extent) == 0,
```

Comment:
Seems like we need to change this too, in order to fix the GPU builds:
```suggestion
        amrex::numParticlesOutOfRange(pti, rho->nGrowVect().min() - shape_extent) == 0,
```

Reply:
Ah forgot that one, thanks!